src/tools/pub/cmd/pub-global.md

```diff
@@ -110,8 +108,42 @@ and passes in two arguments.
 $ pub global run foo:bar arg1 arg2
 {% endprettify %}
 
+
 ### Running a script from your PATH
 
+In order to run a script directly from your path, it must be specified
+under the `executables` tag in its pubspec file.
```

Comment:
I agree with @nex3. It's fine to mention the details in a note, but assuming the package publisher set things up correctly, you shouldn't need to know this. (It should be exception, not the rule.)